libraries/AP_HAL_Linux/RCInput_SBUS.cpp

```diff
@@ -44,56 +80,103 @@ void RCInput_SBUS::set_device_path(const char *path)
     printf("Set SBUS device path %s\n", path);
 }
 
-#ifndef B100000
-// disco uses 010000 for 100000 baud (BOTHER)
-#define B100000 010000
-#endif
+#define SBUS_DEBUG_LOG 0
+#define SBUS_CAUSE_CORRUPTION 0
 
 void RCInput_SBUS::_timer_tick(void)
 {
-    if (device_path == nullptr) {
+    if (fd == -1) {
         return;
     }
-    /*
-      we defer the open to the timer tick to ensure all RPC calls are
-      made in the same thread
-     */
-    if (fd == -1) {
-        fd = open(device_path, O_RDONLY | O_NONBLOCK);
-        if (fd != -1) {
-            printf("Opened SBUS input %s fd=%d\n", device_path, (int)fd);
-            fflush(stdout);
-            struct termios2 tio {};
-
-            int ret = ioctl(fd, TCGETS2, &tio);
-            printf("ret1=%d\n", ret);
-
-            tio.c_iflag &= ~(IGNBRK | BRKINT | PARMRK | ISTRIP | INLCR
-                             | IGNCR | ICRNL | IXON);
-            tio.c_iflag |= (INPCK | IGNPAR);
-            tio.c_oflag &= ~OPOST;
-            tio.c_lflag &= ~(ECHO | ECHONL | ICANON | ISIG | IEXTEN);
-            tio.c_ispeed = B100000;
-            tio.c_ospeed = B100000;
-            tio.c_cflag &= ~(CSIZE | CRTSCTS | PARODD | CBAUD);
-            tio.c_cflag |= (CS8 | CSTOPB | CLOCAL | PARENB | BOTHER | CREAD);
-            tio.c_cc[VMIN] = 25;
-            tio.c_cc[VTIME] = 0;
-            ret = ioctl(fd, TCSETS2, &tio);
-            printf("SBUS ret=%d %s\n", ret, strerror(errno));
+
+    // read up to 10 frames at a time
+    uint8_t bytes[SBUS_FRAME_SIZE*10];
+    int nread;
+    
+    fd_set fds;
+    struct timeval tv;
+    
+    FD_ZERO(&fds);
+    FD_SET(fd, &fds);
+
+    tv.tv_sec = 0;
+    tv.tv_usec = 0;
+
+    // as TMIN is SBUS_FRAME_SIZE the select won't return unless there is
+    // at least SBUS_FRAME_SIZE bytes available
```

Comment:
what exactly this means?    if we have **any** number of bytes available in fd it will return 1.  Where the condition with minimum SBUS_FRAME_SIZE comes from?